server/protocols/webadmin/webadmin-mailrepository/src/main/java/org/apache/james/webadmin/service/MailRepositoryStoreService.java

```diff
@@ -58,18 +62,21 @@ public MailRepositoryStoreService(MailRepositoryStore mailRepositoryStore) {
 
     public List<MailRepositoryResponse> listMailRepositories() {
         return mailRepositoryStore.getUrls()
-            .map(MailRepositoryResponse::new)
+            .map(NormalMailRepositoryResponse::new)
             .collect(Guavate.toImmutableList());
     }
 
     public MailRepository createMailRepository(MailRepositoryUrl repositoryUrl) throws MailRepositoryStore.MailRepositoryStoreException {
         return mailRepositoryStore.create(repositoryUrl);
     }
 
-    public Optional<List<MailKeyDTO>> listMails(MailRepositoryUrl url, Offset offset, Limit limit) throws MailRepositoryStore.MailRepositoryStoreException, MessagingException {
-        Optional<MailRepository> mailRepository = Optional.ofNullable(getRepository(url));
+    public Optional<List<MailKeyDTO>> listMails(MailRepositoryPath path, Offset offset, Limit limit) throws MailRepositoryStore.MailRepositoryStoreException, MessagingException {
         ThrowingFunction<MailRepository, List<MailKeyDTO>> list = repository -> list(repository, offset, limit);
```

Comment:
This method should return a stream, so that you can flatMap it directly and avoid the boiler plate `.flatMap(List::stream)`